mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMailboxInconsistenciesService.java

```diff
@@ -0,0 +1,455 @@
+                fixedInconsistencies = Optional.of(count);
+                return this;
+            }
+
+            public Builder addConflictingEntry(ConflictingEntry conflictingEntry) {
+                conflictingEntries.add(conflictingEntry);
+                return this;
+            }
+
+            public Builder errors(long count) {
+                errors = Optional.of(count);
+                return this;
+            }
+
+            public Context build() {
+                return new Context(
+                    processedMailboxEntries.orElse(0L),
+                    processedMailboxPathEntries.orElse(0L),
+                    fixedInconsistencies.orElse(0L),
+                    conflictingEntries.build(),
+                    errors.orElse(0L));
+            }
+        }
+
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        private final AtomicLong processedMailboxEntries;
+        private final AtomicLong processedMailboxPathEntries;
+        private final AtomicLong fixedInconsistencies;
+        private final ConcurrentLinkedDeque<ConflictingEntry> conflictingEntries;
+        private final AtomicLong errors;
+
+        Context() {
+            this(new AtomicLong(), new AtomicLong(), new AtomicLong(), ImmutableList.of(), new AtomicLong());
+        }
+
+        Context(long processedMailboxEntries, long processedMailboxPathEntries, long fixedInconsistencies, Collection<ConflictingEntry> conflictingEntries, long errors) {
+            this(new AtomicLong(processedMailboxEntries),
+                new AtomicLong(processedMailboxPathEntries),
+                new AtomicLong(fixedInconsistencies),
+                conflictingEntries,
+                new AtomicLong(errors));
+        }
+
+        private Context(AtomicLong processedMailboxEntries, AtomicLong processedMailboxPathEntries, AtomicLong fixedInconsistencies, Collection<ConflictingEntry> conflictingEntries, AtomicLong errors) {
+            this.processedMailboxEntries = processedMailboxEntries;
+            this.processedMailboxPathEntries = processedMailboxPathEntries;
+            this.fixedInconsistencies = fixedInconsistencies;
+            this.conflictingEntries = new ConcurrentLinkedDeque<>(conflictingEntries);
+            this.errors = errors;
+        }
+
+        long getProcessedMailboxEntries() {
+            return processedMailboxEntries.get();
+        }
+
+        long getProcessedMailboxPathEntries() {
+            return processedMailboxPathEntries.get();
+        }
+
+        long getFixedInconsistencies() {
+            return fixedInconsistencies.get();
+        }
+
+        ImmutableList<ConflictingEntry> getConflictingEntries() {
+            return ImmutableList.copyOf(conflictingEntries);
+        }
+
+        long getErrors() {
+            return errors.get();
+        }
+
+        @Override
+        public final boolean equals(Object o) {
```

Comment:
is there a test for that?

Reply:
You are right, comparison makes sens on a POJO, I'll enhance that...